src/crypto/hsm/CHIPCryptoPALHsm.h

```diff
@@ -107,15 +107,31 @@ class P256KeypairHSM : public P256Keypair
 
     bool provisioned_key;
 
-    void SetKeyId(int id) { keyid = id; }
+    void SetKeyId(uint32_t id) { keyid = id; }
 
-    int GetKeyId(void) { return keyid; }
+    uint32_t GetKeyId(void) { return keyid; }
 
 private:
-    int keyid;
+    uint32_t keyid;
 };
 #endif //#if ENABLE_HSM_GENERATE_EC_KEY
 
+#if ENABLE_HSM_PBKDF2_SHA256
+
+class PBKDF2_sha256HSM : public PBKDF2_sha256
+{
+public:
+    uint32_t keyid;
```

Comment:
can this be private and provide an accessor if needed?

Reply:
Sure.  Did as suggested.